### DISCUSSION TITLE
Struggling with Row Level Security in signup flow

### DISCUSSION THREAD

**Discussion by simonh1000** [supabase/supabase#5332] [Questions] [answered]
I have this signup code, that creates a user, and then continues to create a player entry with additional information about the person signing up
```
export async function signUp(credentials: Credentials, name: String): Promise<Player> {
    console.log("signUp", { credentials, name })
    let userRes = await supabase.auth.signUp(credentials);

    if (userRes.error) {
        throw (userRes.error);
    }

    console.log("signUp", userRes)
    const newPlayer = {
        user_id: userRes.user.id,
        name
    }
    // insert takes a list
    const { data, error } = await supabase.from(PLAYERS).insert([newPlayer])
    if (error) return Promise.reject(error)
    return data[0]
}
```

I'm getting an error `'new row violates row-level security policy for table "players"'`

I have the following policies:

![image](https://user-images.githubusercontent.com/5256724/152675675-2c948aa7-3a07-468e-ab02-f7089703536b.png)

What am I missing?

**Comment by GaryAustin1** [comment, marked as answer]
Are you running this code from a browser client or a server?
Check if you have a jwt token in the authorization header in the network call for the insert.

**Reply by GaryAustin1** [reply]
Just to wrap this up.  This was debugged in the discord chat.
If you have email confirmation on, SignUp does NOT return session, so will fail any sort of auth RLS on tables.

A way of sending some extra user data at signUp is to have a trigger function on inserts to auth.users table and write to your public "profile" table using the user meta data offered in signUp.

**Comment by simonh1000** [comment]
Yes, I do have the token set - I'm using the SDK.
I've been debugging in the SQL editor and the following is what it took to get an insert to work
```
SET role authenticated;
SELECT set_config('request.jwt.claims', jsonb_build_object('role', 'authenticated')::jsonb::text, true)::text; 

insert into players (user_id, name) values ('....', 'some_name');
```
But I have no idea how to emulate that using the SDK
